keyless/circuit-tests/src/packing.rs

```diff
@@ -0,0 +1,159 @@
+
+        let circuit_input_signals = CircuitInputSignals::new()
+            .byte_input("num_in", n)
+            .bytes_input("bits_out", &expected_num2bits_be(n as u64, bits_max_size))
+            .pad(&config).unwrap();
+
+        let result = circuit_handle.gen_witness(circuit_input_signals);
+        println!("{:?}", result);
+        assert!(result.is_ok());
+    }
+}
+
+#[test]
+fn bits2num_big_endian_test() {
+    let circuit_handle = TestCircuitHandle::new("packing/bits2num_big_endian_test.circom").unwrap();
+    let mut rng = ChaCha20Rng::seed_from_u64(TEST_RNG_SEED);
+    let bits_max_size = 64;
+
+    for i in 0..=255 {
+        let expected_n = rng.next_u64();
+        let bits_in = expected_num2bits_be(expected_n, bits_max_size);
+
+        let config = CircuitPaddingConfig::new()
+            .max_length("bits_in", bits_max_size);
+
+        let circuit_input_signals = CircuitInputSignals::new()
+            .bytes_input("bits_in", &expected_num2bits_be(expected_n, bits_max_size))
+            .u64_input("num_out", expected_n)
+            .pad(&config).unwrap();
+
+        let result = circuit_handle.gen_witness(circuit_input_signals);
+        println!("{:?}", result);
+        assert!(result.is_ok());
+   }
+}
+
+#[test]
+fn bytes_to_bits_test() {
+    let circuit_handle = TestCircuitHandle::new("packing/bytes_to_bits_test.circom").unwrap();
+
+    let mut rng = ChaCha20Rng::seed_from_u64(TEST_RNG_SEED);
+    const bytes_max_size : usize = 10;
+    const bits_max_size : usize = bytes_max_size * 8;
+
+
+    for i in 0..=255 {
+        let bytes : &mut [u8] = &mut [0u8; bytes_max_size];
+        rng.fill_bytes(bytes);
+
+        let expected_bits : Vec<u8> = bytes
+            .into_iter()
+            .map(|byte| expected_num2bits_be(*byte as u64, 8))
+            .flatten()
+            .collect();
+
+        let config = CircuitPaddingConfig::new()
+            .max_length("bytes_in", bytes_max_size)
+            .max_length("bits_out", bits_max_size);
+
+        let circuit_input_signals = CircuitInputSignals::new()
+            .bytes_input("bytes_in", bytes)
+            .bytes_input("bits_out", &expected_bits)
+            .pad(&config).unwrap();
+
+        let result = circuit_handle.gen_witness(circuit_input_signals);
+        println!("{:?}", result);
+        assert!(result.is_ok());
+    }
+}
+
+#[test]
+fn bits_to_field_elems_test() {
+    let circuit_handle = TestCircuitHandle::new("packing/bits_to_field_elems_test.circom").unwrap();
+
+    let mut rng = ChaCha20Rng::seed_from_u64(TEST_RNG_SEED);
+
+    // Keeping both of these constant makes sense here, since the circuit only ever uses these two
+    // particular parameters when calling this template
+    const max_bits_len : usize = 256;
+    const bits_per_field_elem : usize = 64;
+    let num_field_elems = if max_bits_len % bits_per_field_elem == 0 { max_bits_len / bits_per_field_elem } else { (max_bits_len / bits_per_field_elem) + 1 };
```

Comment:
nit: can be simplified to
```suggestion
    let num_field_elems = max_bits_len.div_ceil(bits_per_field_elem);
```
?

Reply:
yes, thank you! Didn't know about that method